### ISSUE TITLE
It would be nice if MonadRandom did not depend on System.Random

### ISSUE THREAD

**Issue by jeffreyrosenbluth** [byorgey/MonadRandom#31]
In particular if the IO instance was not tied to `randomIO` and `stdGen`.
So that we could plug in other prng's like tf-random for things like `evalRandIO`.

**Comment by tom-bop** [comment]
Does this mean there's currently no way to use `MonadRandom` and avoid the wonky behavior of http://kenta.blogspot.co.nz/2014/02/rzzllapd-nonrandomness-of-first-random.html ?

**Comment by tom-bop** [comment]
Hm, appears we can at least work around it(?). Sorry for the noise if so.

```
>  :m + Control.Monad.Random
>  :m + System.Random.TF
>  (fst . randomR (0::Int, 7)) <$> newTFGen
5
```

**Comment by lehins** [comment]
All these issues has been fixed in `random-1.2`. 

`IO` instance for `MonadRandom` is a limitation of design IMHO, which has also been solved with `StatefulGen` in `random-1.2`
